### ISSUE TITLE
This docker is pure magic 

### ISSUE THREAD

**Issue by chaos0frenzy** [Steam-Headless/docker-steam-headless#36]
### What is your feature request?

This might not be the place for this but I wanted to thank you for all your work this is the coolest docker I have ever seen. I need you to know it’s greatly appreciated.

**Comment by VoxTrek** [comment]
Thank you so much for this docker container!
I want to mention how good the quality is of this docker container.
My homelab is Dell T150, nVidia T400 ;-) , 300 Watt PSU, 72 GB ECC RAM. 25TB ZFS. Proxmox and docker. Plex & Jellyfin with HW encoding.

Here is my docker-compose.yml file:
Only had to add runtime: nvidia to force the docker container to install nvidia gpu drivers.
The sound worked directly via the web NOVNC access or via Steam client.

version: '3'

services:
  steam-headless:
    container_name: steam
    image: josh5/steam-headless:latest
    restart: unless-stopped
    runtime: nvidia
    ## NOTE: Requires privileged access to host to be able to access the required devices
    privileged: true
    shm_size: 2GB
    ipc: host # Could also be set to 'shareable'
    ulimits:
      nofile:
        soft: 1024
        hard: 524288

    # NETWORK:
    ## NOTE:  Steam headless always requires the use of the host network.
    ##        If we do not use the host network, then device input is not possible
    ##        and your controllers will not work in steam games.
    network_mode: host
    hostname: steam
    extra_hosts:
      - "dell:192.168.0.187"
    
    # ENVIRONMENT:
    ## Read all config variables from the .env file
    #env_file: .env

    # VOLUMES:
    volumes:
      # The location of your home directory.
      - /shared/docker/steam/home/:/home/default/:rw
      # The location where all games should be installed.
      - /shared/docker/steam/games/:/mnt/games/:rw
      # Input devices used for mouse and joypad support inside the container.
      - /dev/input/:/dev/input/:ro
      # The Xorg socket. This will be shared with other containers so they can access the X server.
      #- /opt/container-data/steam-headless/.X11-unix/:/tmp/.X11-unix/:rw
      # Pulse audio socket. This will be shared with other containers so they can access the audio sink.
      #- /opt/container-data/steam-headless/pulse/:/tmp/pulse/:rw

Now we can play for example Half-life 2 with more than 60 fps running on the server streaming to a Steam client.
Always nice to have an extra game system in the house.

**Comment by Crydsch** [comment]
I too, want to thank you for this project!

Having tried a lot of other things to get remote gaming to work .. and failing with all of them.
This was the last one I found and to my surprise - it just works:tm:   :D

I am using this with a nvidia GPU and even the hardware acceleration just worked out of box.
Now I can game on a very old low powered laptop by using the already existing GPU server.

Thank you so much! <3

**Comment by stuCONNERS1983** [comment]
This is an amazing docker container. Works fantastic on Open Media Vault with full GPU access. Thank-you for your hard work.

**Comment by Josh5** [comment]
It is so cool to hear feedback like this. This absolutely makes my day!

**Comment by tnielsen2** [comment]
Yes, thank you very much for your work on this project. I sent a coffee your way as a way to say thanks!

**Comment by ponlayookm** [comment]
I'm using Azure's GRID driver, and I think the script's downloaded normal 535 and makes X11 crash everytime.
In order for Tesla T4 cards to operate in 3D graphic mode, it needs GRID driver.
https://www.youtube.com/watch?v=5nLhKhnQRjo

**Comment by Maxoobidou** [comment]
I can install it yet (waiting for it to be available on Truenas), but I wanted to thank you too for taking on the challenge.
This looks awesome and I can't wait to try it !

**Comment by Josh5** [comment]
> I can't install it yet (waiting for it to be available on Truenas), but I wanted to thank you too for taking on the challenge.
> This looks awesome and I can't wait to try it !

Truenas scale?

**Comment by Maxoobidou** [comment]
Yes !
They've been really growing the official community apps repo lately, would be cool if Steam-headless could get on there :)
If not, Truecharts is a third-party repo that is easy to use too.

**Comment by Josh5** [comment]
> Yes !
> They've been really growing the official community apps repo lately, would be cool if Steam-headless could get on there :)
> If not, Truecharts is a third-party repo that is easy to use too.

Would you consider sponsoring a TrueNAS Scale development box?

**Comment by Maxoobidou** [comment]
> > Yes !
> > They've been really growing the official community apps repo lately, would be cool if Steam-headless could get on there :)
> > If not, Truecharts is a third-party repo that is easy to use too.
> 
> Would you consider sponsoring a TrueNAS Scale development box?

Yes ! Sorry for the late reply, didn't catch the message.
What would that entails ? Truenas scale can be installed on pretty much anything. I can also do tests on my side if needed, I have an old AMD card lying around somewhere.

**Comment by Josh5** [comment]
> As many others mentioned already. I would like to say THANKS (caps) to you for doing this. I have a TrueNas with an NVIDIA card and recently found this project on truecharts. ![image](https://private-user-images.githubusercontent.com/400238/312859912-7dc288db-9302-4fa6-b861-408c129aabe7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTEzMTYyNDMsIm5iZiI6MTcxMTMxNTk0MywicGF0aCI6Ii80MDAyMzgvMzEyODU5OTEyLTdkYzI4OGRiLTkzMDItNGZhNi1iODYxLTQwOGMxMjlhYWJlNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMyNFQyMTMyMjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZWQ2YTdhMTZiYzlhMTQzMTU0YmZkMTdhYzVlMjMyYjg5Zjc0MTcxZmNiOGI5N2JhOTViNGU1YzAzM2U3NmRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.cGWfB9gE9b4Twi5i4gC5a1jir0yV32aDpn4q4ytUiOU)
> 
> I would just like to reiterate the creation of documentation for anyone who would like to use this project with TrueNas. I would be interested in being part of that documentation if needed.
> 
> Thanks a lot for your hard work.

Awesome. 
I am thinking about creating a docs site for this project soon. Help with this TrueNAS documentation would be very welcome.

**Comment by Fuckingnameless** [comment]
> I'm using Azure's GRID driver, and I think the script's downloaded normal 535 makes X11 crash every time. In order for Tesla T4 cards to operate in 3D graphic mode, it needs GRID driver. https://www.youtube.com/watch?v=5nLhKhnQRjo

do you know if tesla pascal needs grid drivers?

**Comment by ntn888** [comment]
My cheap projector has a drastic bad delay via hdmi, streaming was the alternative.. This package makes the setup so much easy.. this software is so undervalued.. everything works and stable!

I came across it in reddit when searching for my solutions. the documentation has some catchup to do though. I guess it's intended usecase of remote portable clients is mindblowing too :)

**Comment by Josh5** [comment]
Just closing this as a cleanup of all the issues as I prepare for the v2 release.
